plugins/module_utils/query.py

```diff
@@ -0,0 +1,225 @@
+        "<",
+        "<=",
+        ">",
+        ">=",
+        "BETWEEN",
+        "DATEPART",
+        "RELATIVEGE",
+        "RELATIVELE",
+        "RELATIVEGT",
+        "RELATIVELT",
+        "RELATIVEEE",
+        "ISEMPTY",
+        "ISNOTEMPTY",
+        "ANYTHING",
+        "SAMEAS",
+        "NSAMEAS",
+        "MORETHAN",
+        "LESSTHAN",
+    )
+)
+OPERATORS_NUMERIC = set(
+    (
+        "=",
+        "!=",
+        "EMPTY",
+        "NOTEMPTY",
+        "<",
+        ">",
+        "<=",
+        ">=",
+        "BETWEEN",
+        "ANYTHING",
+        "SAMEAS",
+        "NSAMEAS",
+        "GT_FIELD",
+        "LT_FIELD",
+        "GT_OR_EQUALS_FIELD",
+        "LT_OR_EQUALS_FIELD",
+    )
+)
+OPERATORS_BOOLEAN = set(
+    (
+        "=",
+        "!=",
+        "ISEMPTY",
+        "ISNOTEMPTY",
+        "ANYTHING",
+        "SAMEAS",
+        "NSAMEAS",
+    )
+)
+OPERATORS_EMAIL = set(("VALCHANGES", "CHANGESFROM", "CHANGESTO"))
+
+OPERATORS = (
+    OPERATORS_STRING
+    | OPERATORS_REFERENCE
+    | OPERATORS_CHOICE_STRINGS
+    | OPERATORS_CHOICE_INTEGERS
+    | OPERATORS_DATE_TIME
+    | OPERATORS_NUMERIC
+    | OPERATORS_BOOLEAN
+    | OPERATORS_EMAIL
+)
+
+
+def _insert_and(queries):
+    return "^".join("{0}".format(i) for i in queries)
+
+
+def _insert_or(queries):
+    return "^NQ".join("{0}".format(i) for i in queries)
+
+
+def get_operator_and_value(operator):
+    # Return operator and value
+    for o in OPERATORS:
+        if operator.startswith(o) and operator[len(o)] == " ":
+            return (o, operator[len(o) + 1 :])
+
+    return None, None
+
+
+def pars_queries(queries):
+    # input: list({"caller": "= abel.tuter", "state": "= new"}, {"short_description": "LIKE SAP"})
+    # output: list({"caller": ("=", "able.tuter"), "state": ("=", "new")}, {"short_description": ("LIKE", "SAP")}})
+    parsed_queries = []
+    errors = []
+
+    for item in queries:
+        query = dict()
+        for collumn, value in item.items():
+            oper, field = get_operator_and_value(value)
+            if oper:
+                query[collumn] = (oper, field)
+            else:
+                errors.append(
+                    "Invalid operator {0} for query in collumn {1}.".format(
+                        value, collumn
+                    )
+                )
+        if query:
+            parsed_queries.append(query)
+
+    return parsed_queries, errors
+
+
+def serialize_query(parsed_queries):
+    # input: list({"caller_id": ("=", "712083754987"), "state": ("=", "3")}, {})
+    # output: "caller_id=712083754987^state=3"
+    or_queries = []
+
+    for item in parsed_queries:
+        and_queries = []
+        for collumn, value in item.items():
+            and_queries.append(collumn + value[0] + value[1])
+        # isert AND between items
+        or_queries.append(_insert_and(and_queries))
+
+    # insert OR between items
+    return _insert_or(or_queries)
+
+
+def fields_mapping(module, queries):
```

Comment:
Utility functions should not use the AnsibleModule instance (instances should not "leave" the `plugins/modules` folder).

```suggestion
def map_query_values(query, mapper):
```